### PR TITLE
Update GA to version 4

### PR DESCRIPTION
**Pull request by MarekB01**
https://github.com/BroadleafCommerce/QA/issues/4848

Updated to GA4.
Created a custom thymeleaf processor for new version of GA

### CODE REVIEW

File: core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/GoogleAnalytics4Processor.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    protected String getDisplayAdvertisingJs(String trackerPrefix) {
+        return "gtag('" + trackerPrefix + "require', 'displayfeatures');";
+    }
+
+    /**
+     * Builds the transaction analytics for the given tracker name. Invokes {@link #getItemJs(Order, String) for each item
+     * in the given <b>order</b>.
+     */
+    protected String getTransactionJs(Order order, String trackerPrefix) {
+        StringBuffer sb = new StringBuffer();
```
Comment (oleksiimiroshnyk) [inline core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/GoogleAnalytics4Processor.java]: guess StringBuilder is better ? Buffer is synch which is not needed here

File: core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/GoogleAnalytics4Processor.java [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    }
+
+    @Override
+    public int getPrecedence() {
+        return 0;
+    }
+
+    @Override
+    public BroadleafTemplateModel getReplacementModel(String tagName, Map<String, String> tagAttributes, BroadleafTemplateContext context) {
+        StringBuffer sb = new StringBuffer();
```
Comment (oleksiimiroshnyk) [inline core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/GoogleAnalytics4Processor.java]: builder instead of buffer

**Review by oleksiimiroshnyk — CHANGES_REQUESTED** [review]

File: core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/GoogleUniversalAnalyticsProcessor.java [hunk 3]
```diff
@@ -111,10 +115,12 @@ public BroadleafTemplateModel getReplacementModel(String tagName, Map<String, St
         StringBuffer sb = new StringBuffer();
```
Comment (oleksiimiroshnyk) [inline core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/GoogleUniversalAnalyticsProcessor.java]: one more buffer

**Review by oleksiimiroshnyk — APPROVED** [review]

**Comment by riteshadhikari17** [comment]
@MarekB01 @oleksiimiroshnyk I don't know how to test this. Have you guys done any testing? If you guys are confident about this, you can simply merge it.

**Comment by Jerry77oz** [comment]
@MarekB01 this looks like we're good to merge this, anything you think we could test before we do?

**Comment by oleksiimiroshnyk** [comment]
used docs for migration:
https://developers.google.com/analytics/devguides/migration/ecommerce/analyticsjs-before-you-begin\
https://developers.google.com/analytics/devguides/migration/ecommerce/gtagjs-ua-to-ga4
https://support.google.com/analytics/answer/9310895#gtagjs-enable-basic&zippy=%2Cin-this-article
https://support.google.com/analytics/answer/10759417
https://support.google.com/analytics/answer/9744165#zippy=%2Cin-this-article

new property for tag id: googleAnalytics4.tagIdForProperty

**Pull request merged**